app/models/manageiq/providers/microsoft/infra_manager/refresh_parser.rb

```diff
@@ -328,6 +328,7 @@ def process_host_hardware(host)
         :cpu_total_cores      => host['LogicalProcessorCount'],
         :cpu_cores_per_socket => host['CoresPerCPU'],
         :guest_devices        => process_host_guest_devices(host),
+        :serial_number        => host['SMBiosGUIDString'],
```

Comment:
If this is null (and I noticed that on one of our 3 dev hosts it is null), I think you'll end up with a blank string in the database because of the `-As [string]` stuff. If memory serves, PostgreSQL will retain an empty string (while MySQL will treat it as null).

So, we might want to do:

`host['SMBiosGUIDString'].blank? ? nil : host['SMBiosGUIDString']`

Or does Rails intervene here?

Reply:
Note that I could be wrong here. Perhaps the `ConvertTo-Json` call effectively treats empty strings as nulls. Something to doublecheck anyway.

Reply:
Ah it was nil when I was getting `host['PhysicalMachine']['SMBiosGUID']` but you're right now its an empty string